ros/src/mux_demux/scripts/mux_demux.py

```diff
@@ -1,60 +1,84 @@
 #! /usr/bin/python
 import rospy
+import json
+import packet_mapper
 from shared_msgs.msg import can_msg, auto_command_msg, thrust_status_msg, thrust_command_msg, esc_single_msg
 from sensor_msgs.msg import Imu, Temperature
 from std_msgs.msg import Float32
 
-def thrust_status_received(msg):
-  pass # This runs on a seperate thread
-def esc_single_received(msg):
-  pass # This runs on a seperate thread
+from threading import Lock
+from flask import Flask, render_template, session, request
+from flask_socketio import SocketIO, emit, disconnect
 
-def temp_received(msg):
-  pass # This runs on a seperate thread
-def imu_received(msg):
-  pass # This runs on a seperate thread
-def ph_received(msg):
-  pass # This runs on a seperate thread
-def depth_received(msg):
-  pass # This runs on a seperate thread
+async_mode = None
+app = Flask(__name__)
+app.config['SECRET_KEY'] = 'key'
+socketio = SocketIO(app, async_mode=async_mode)
+thread = None
+thread_lock = Lock()
+
+with open ('../../../../surface/frontend/src/packets.json') as json_data:
+  data = json.load(json_data,)
+
+dearFlask = data['dearflask']
+dearClient = data['dearclient']
+flaskMapper = packet_mapper.packet_mapper(dearFlask)
+clientMapper = packet_mapper.packet_mapper(dearClient)
+thrust_pub = None
+auto_pub = None
```

Comment:
I told you to look up the 'global' keyword to use. Otherwise I don't think your socketio.on() callbacks are going to find auto_pub or dearFlask.